Util/src/main/java/io/deephaven/util/HeapDump.java

```diff
@@ -26,51 +24,18 @@ public static void heapDump() throws IOException {
     }
 
     public static String generateHeapDumpPath() {
-        final Configuration configuration = Configuration.getInstance();
-        final String processName = configuration.getProcessName();
-        return configuration.getLogPath(processName + "_"
-                + new SimpleDateFormat("yyyyMMddHHmmss").format(new Date(System.currentTimeMillis())) + ".hprof");
+        final String name =
+                new SimpleDateFormat("yyyyMMddHHmmss").format(new Date(System.currentTimeMillis())) + ".hprof";
+        final Path path = DataDir.get().resolve("heapDumps").resolve(name);
+        return path.toString();
     }
 
     @SuppressWarnings("WeakerAccess")
     public static void heapDump(String filename) throws IOException {
+        Files.createDirectories(Path.of(filename).getParent());
         final MBeanServer server = ManagementFactory.getPlatformMBeanServer();
         final HotSpotDiagnosticMXBean mxBean = ManagementFactory.newPlatformMXBeanProxy(server,
                 "com.sun.management:type=HotSpotDiagnostic", HotSpotDiagnosticMXBean.class);
         mxBean.dumpHeap(filename, true);
     }
-
-    private static void heapDumpWrapper(final String cause, final RuntimeException failure,
-            final Predicate<RuntimeException> ignore, final Logger log) {
-        if (ignore != null && ignore.test(failure)) {
-            return;
-        }
-        try {
-            final String heapDumpPath = HeapDump.generateHeapDumpPath();
-            log.fatal().append(cause + ", generating heap dump to")
-                    .append(heapDumpPath).append(": ").append(failure).endl();
-            heapDump(heapDumpPath);
-        } catch (Exception e) {
-            log.info()
-                    .append("Exception while trying to dump heap on assertion failure: " + e.getMessage() + ":\n")
-                    .append(e)
-                    .endl();
-        }
-    }
-
-    public static void setupHeapDumpWithDefaults(final Configuration configuration,
```

Comment:
Can we put this back and actually enable it from `Main`?

Reply:
Will do.